From b4712481174f2114d26d4404b5cdf67209db16bc Mon Sep 17 00:00:00 2001
From: Stefan Lellwitz <lel@iesy.com>
Date: Wed, 24 Aug 2022 14:27:21 +0200
Subject: [PATCH 1/2] drivers: drm: bridge: lt8912: make sure to use device
 tree for setting MIPI P/N lines swap

MIPI P/N lines should not be swapped because with add-on PCB (ZP376) signals are not crossed anymore.

Signed-off-by: Stefan Lellwitz <lel@iesy.com>
---
 drivers/gpu/drm/bridge/lt8912-i2c.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/bridge/lt8912-i2c.c b/drivers/gpu/drm/bridge/lt8912-i2c.c
index 3963bcf455c5..3ae71e553606 100644
--- a/drivers/gpu/drm/bridge/lt8912-i2c.c
+++ b/drivers/gpu/drm/bridge/lt8912-i2c.c
@@ -125,6 +125,7 @@ struct lt8912 {
 	int hdmi_y;    //0x00:RGB, 0x01:YCbCr422 0x02:YCbCr 444
     u8 sink_edid[256];
 	u8 audio_mode;					/* selected audio mode - valid only for HDMI output */
+	bool swap_mipi_pn;
 };
 
 
@@ -343,13 +344,14 @@ static void lt8912_init(struct lt8912 *lt)
 	regmap_write(lt->regmap[0], 0x5a, 0x02);
 
 	/* MIPIAnalog */
-	//if(lt->swap_mipi_pn) {
-	if (true) {
+	if(lt->swap_mipi_pn) {
 		pr_debug("lt8912_init(): MIPI P/N lines swapped\n");
 
 		// P/N mipi lines are swapped, lp rx trigger point 700mV
 		regmap_write(lt->regmap[0], 0x3e, 0xf6); 
 	} else {
+		pr_debug("lt8912_init(): MIPI P/N lines NOT swapped\n");
+
 		// P/N mipi lines not swapped, lp rx trigger point 500mV
 		regmap_write(lt->regmap[0], 0x3e, 0xce); 
 	}
@@ -579,6 +581,8 @@ static int lt8912_probe(struct i2c_client *client, const struct i2c_device_id *i
 		return ret;
 	}
 
+	lt8912->swap_mipi_pn = of_property_read_bool(dev->of_node, "swap-mipi-pn");
+
 	ret = lt8912_i2c_init(lt8912, client);
 	if (ret)
 		return ret;
-- 
2.30.2

